data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -539,6 +541,8 @@ private <K> K readEntity(RS rs, MappingContext<K> ctx, @Nullable Object parent,
                                     if (associatedEntity != null) {
                                         Objects.requireNonNull(associatedId);
                                         joinCtx.associate(associatedCtx, associatedId, associatedEntity);
+                                    } else {
+                                        joinCtx.manyAssociations = new HashMap<>();
```

Comment:
This is needed in case one to many returns more than one item, if we don't initialize here in case when first item is null (null id) it will skip reading following items too.